cpp/tests/lists/contains_tests.cpp

```diff
@@ -1051,19 +1046,781 @@ TYPED_TEST(TypedContainsDecimalsTest, VectorKey)
   {
     // CONTAINS
     auto result   = lists::contains(search_space->view(), search_key->view());
-    auto expected = bools{1, 0, 0, 1, 1, 0, 0, 0, 1, 0};
+    auto expected = bools_col{1, 0, 0, 1, 1, 0, 0, 0, 1, 0};
     CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(expected, *result);
   }
   {
     // FIND_FIRST
     auto result   = lists::index_of(search_space->view(), search_key->view(), FIND_FIRST);
-    auto expected = indices{1, absent, absent, 2, 0, absent, absent, absent, 2, absent};
+    auto expected = indices_col{1, ABSENT, ABSENT, 2, 0, ABSENT, ABSENT, ABSENT, 2, ABSENT};
     CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(expected, *result);
   }
   {
     // FIND_LAST
     auto result   = lists::index_of(search_space->view(), search_key->view(), FIND_LAST);
-    auto expected = indices{1, absent, absent, 2, 0, absent, absent, absent, 2, absent};
+    auto expected = indices_col{1, ABSENT, ABSENT, 2, 0, ABSENT, ABSENT, ABSENT, 2, ABSENT};
+    CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(expected, *result);
+  }
+}
+
+template <typename T>
+struct TypedStructContainsTest : public ContainsTest {
+};
+TYPED_TEST_SUITE(TypedStructContainsTest, ContainsTestTypes);
+
+TYPED_TEST(TypedStructContainsTest, EmptyInputTest)
+{
+  using tdata_col = cudf::test::fixed_width_column_wrapper<TypeParam, int32_t>;
+
+  auto const lists = [] {
+    auto offsets = indices_col{};
+    auto data    = tdata_col{};
+    auto child   = structs_col{{data}};
+    return make_lists_column(0, offsets.release(), child.release(), 0, {});
+  }();
+
+  auto const scalar_key = [] {
+    auto child = tdata_col{0};
+    return make_struct_scalar(child);
+  }();
+  auto const column_key = [] {
+    auto child = tdata_col{};
+    return structs_col{{child}};
+  }();
+
+  auto const result1  = lists::contains(lists->view(), scalar_key);
+  auto const result2  = lists::contains(lists->view(), column_key);
+  auto const expected = bools_col{};
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(expected, *result1);
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(expected, *result2);
+}
+
+TYPED_TEST(TypedStructContainsTest, ScalarKeyNoNullLists)
+{
+  using tdata_col = cudf::test::fixed_width_column_wrapper<TypeParam, int32_t>;
+
+  auto const lists = [] {
+    auto offsets = indices_col{0, 4, 7, 10, 15, 18, 21, 24, 24, 28, 28};
+    // clang-format off
+    auto data1    = tdata_col{0, 1, 2, 1,    //
```

Comment:
Are the trailing `//` necessary if clang-format is already turned off in this section? I'd remove the trailing `//`.